gc/base/ObjectScanner.hpp

```diff
@@ -309,6 +309,8 @@ class GC_ObjectScanner : public MM_BaseVirtual
 
 	MMINLINE bool hasMoreSlots() { return 0 == (GC_ObjectScanner::noMoreSlots & _flags); }
 
+	MMINLINE bool isEmpty() { return (0 == _scanMap) && !hasMoreSlots(); }
```

Comment:
I would expect _scanMap to only be 0 is hasMoreSlots returns false.  Can you explain why this is not true?

Reply:
Large objects (and split array segments) may contain >64 object references and will require >1 scan map to feed to object scanner. The _scanMap == 0 condition will occur with hasMoreSlots(true) for each scan map but the last after each scan map is exhausted.